### ISSUE TITLE
11-image-classifier-dogs-vs-cats.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/josefina-aispuro-merelles-machine-learning#7]
### LO QUE HAS HECHO BIEN

- **Extracción del ZIP**: Código completo para descomprimir el dataset.
- **Split train/test manual**: 80/20 usando `train_test_split` con `random_state=42` - reproducible.
- **Organización de carpetas**: Estructura correcta train/test con subcarpetas por clase.
- **Visualización**: Primeras 9 imágenes de perros y gatos.
- **Normalización**: `rescale=1./255` correcto.
- **ImageDataGenerator**: Configurado correctamente con `target_size`, `batch_size`, `class_mode`.
- **Arquitectura CNN**: Conv2D → Conv2D → Conv2D → MaxPool → Flatten → Dense → Dense.
- **EarlyStopping**: Callback implementado con patience=2.
- **Guardado del modelo**: `model.save()` al final.
- **Conclusiones**: Interpretación del resultado.

---

#### 1. **Inconsistencia class_mode vs loss function - ERROR**

Cuando haces clasificación binaria (perro vs gato = 2 clases), tienes dos opciones,  OPCION A la binaria pura (1 neurona) o OPCÓN B la categórica (2 neuronas).
Has usado y mezclado las dos.

```python
class_mode="binary"  # Genera etiquetas 0 y 1 - OPCION A
# PERO abajo usas la OPCION B meclando el modelo lo que hace que esté mal configurado...
model.add(Dense(units=2, activation="softmax"))  # 2 neuronas de salida
model.compile(loss="sparse_categorical_crossentropy")  # Loss para múltiples clases
```

**El problema**: Usas `class_mode="binary"` que genera etiquetas simples (0 o 1), pero tu modelo tiene 2 neuronas de salida con softmax y loss `sparse_categorical_crossentropy`. 

**Solución A** (más común para binario):
```python
model.add(Dense(units=1, activation="sigmoid"))
model.compile(loss="binary_crossentropy", ...)
```

**Solución B** (si quieres 2 salidas):
```python
class_mode="categorical"  # o "sparse"
```

#### 2. **Solo 1 MaxPooling después de 3 Conv2D**
La arquitectura es inusual. Lo típico es alternar Conv2D con MaxPooling:
```python
Conv2D → MaxPool → Conv2D → MaxPool → Conv2D → MaxPool → Flatten
```

#### 3. **Métricas hardcodeadas en la visualización**
```python
train_accuracy = [0.6122, 0.7505, 0.8543, 0.9607]  # Valores manuales
```
Deberías usar el historial del entrenamiento:
```python
history = model.fit(...)
plt.plot(history.history['accuracy'])
```

#### 4. **Falta normalizar la imagen de predicción**
```python
img = np.asarray(img)  # Valores 0-255
# Falta:
img = img / 255.0  # Normalizar como en entrenamiento
```

#### 5. **Falta evaluación con métricas**
Importas `confusion_matrix` y `classification_report` pero no los usas. Falta:
```python
loss, acc = model.evaluate(test_generator)
```

#### 6. **Rutas absolutas hardcodeadas**

Mejor usar rutas relativas.

--

Celda no ejecutada:

<img width="953" height="527" alt="Image" src="https://github.com/user-attachments/assets/71e473a3-366b-477a-92cc-9e8dec8e80dc" />

---

@josefina-aispuro-merelles , el notebook tiene la estructura básica pero hay un error conceptual importante: la inconsistencia entre `class_mode="binary"` y la arquitectura con 2 salidas + `sparse_categorical_crossentropy`. También faltaría evaluación con métricas reales y normalizar la imagen antes de predecir. 

**Correcciones necesarias:**
1. Corregir class_mode/loss/arquitectura para que sean consistentes
2. Usar `history.history` para las gráficas
3. Normalizar imagen antes de predecir (`img/255.0`)
4. Añadir `model.evaluate()` con métricas
